whylog/config/parser_name_manager.py

```diff
@@ -0,0 +1,48 @@
+import re
+
+from ordered_set import OrderedSet
+
+
+class ParserNameManager(object):
+    def __init__(self, parsers):
+        self._parsers_name = OrderedSet()
+        for parser_name in parsers:
+            self._parsers_name.add(parser_name)
+
+    def add_new_parser_name(self, parser_name):
+        self._parsers_name.add(parser_name)
+
+    def is_free_parser_name(self, parser_name, black_list):
+        return (parser_name not in self._parsers_name) and (parser_name not in black_list)
+
+    def propose_parser_name(self, line, regex_str, black_list):
+        building_words = ParserNameManager._get_building_words(line, regex_str)
+        if not building_words:
+            return self._find_free_by_number_appending('parser_name', black_list)
+        if len(building_words) == 1:
+            if self.is_free_parser_name(building_words[0], black_list):
+                return building_words[0]
+            else:
+                return self._find_free_by_number_appending(building_words[0], black_list)
+        propsed_name = building_words[0] + '_' + building_words[1]
+        if self.is_free_parser_name(propsed_name, black_list):
+            return propsed_name
+        return self._find_free_by_number_appending(
+            building_words[0] + '_' + building_words[1], black_list
```

Comment:
`propsed_name = building_words[0] + '_' + building_words[1]`
so change this line to:
`propsed_name, black_list`